File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result = []
        max_can = candies[0]
        for i in candies:
            if max_can < i:
                max_can = i 
        for i in candies:
            cur = i + extraCandies
            if cur >= max_can:
                result.append(True)
            else:
                result.append(False)
        return result

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_can = *max_element(candies.begin(),candies.end());
        vector<bool> result;
        for(auto c : candies) {
            result.push_back(c + extraCandies >= max_can);
        }
        return result;
    }
};